source/backend/api/Services/ProjectService.cs

```diff
@@ -134,44 +136,107 @@ public PimsProject Add(PimsProject project)
                 throw new ArgumentNullException(nameof(project), "Project cannot be null.");
             }
 
-            // TODO: Verify this is not needed
-            //CheckForDuplicateProducts(project.PimsProducts, project.Id);
+            var externalProducts = MatchProducts(project);
+            if (externalProducts.Count > 0 && !userOverrides.Contains(UserOverrideCode.ProductReuse))
+            {
+                var names = externalProducts.Select(x => $"{x.Code} {x.Description}");
+                throw new UserOverrideException(UserOverrideCode.ProductReuse, $"The product(s) {string.Join(",", names)} can also be found in one or more other projects. Please verify the correct product is being added");
+            }
 
             var newProject = _projectRepository.Add(project);
             _projectRepository.CommitTransaction();
 
             return _projectRepository.Get(newProject.Internal_Id);
         }
 
-        public PimsProject Update(PimsProject project)
+        public PimsProject Update(PimsProject project, IEnumerable<UserOverrideCode> userOverrides)
         {
             _user.ThrowIfNotAuthorized(Permissions.ProjectEdit);
             project.ThrowIfNull(nameof(project));
             _logger.LogInformation($"Updating project with id ${project.Internal_Id}");
 
-            // TODO: Verify this is not needed
-            //CheckForDuplicateProducts(project.PimsProducts, project.Id);
+            var externalProducts = MatchProducts(project);
+            if (externalProducts.Count > 0 && !userOverrides.Contains(UserOverrideCode.ProductReuse))
+            {
+                var names = externalProducts.Select(x => $"{x.Code} {x.Description}");
+                throw new UserOverrideException(UserOverrideCode.ProductReuse, $"The product(s) {string.Join(",", names)} can also be found in one or more other projects. Please verify the correct product is being added");
+            }
+
             var updatedProject = _projectRepository.Update(project);
+
             AddNoteIfStatusChanged(project);
             _projectRepository.CommitTransaction();
 
             return updatedProject;
         }
 
-        /*private void CheckForDuplicateProducts(IEnumerable<PimsProduct> products, long projectId)
+        /*
+        Updates the passed project with the matched products in place.
+        Note: The return list contains the external products matched
+        */
+        private List<PimsProduct> MatchProducts(PimsProject project)
```

Comment:
This function does seem like one we should have good test coverage on (via the public method).